### DISCUSSION TITLE
Steps /degree for Rotary Axis

### DISCUSSION THREAD

**Discussion by Gabjohn1** [terjeio/ioSender#281] [Q&A]
Hi
I am trying to get the correct steps/degree value for my A Rotary Axis.
I have a 200 / 1.8 degree motor 6:1 ratio
So to start I used this equation  (not sure if this is correct):
(200*8*6)/360   = 26.6667  steps/degree.
This value was pretty accurate with a 115mm diameter workpiece. My final $103 = 26.78  IE. I jogged 200mm and measured right at 200mm on the workpiece.
But when I put in a smaller diameter workpiece this value obviously doesn’t work.
Should the steps/ degree be the same for any diameter workpiece?  I guess somehow I am not telling IO sender the starting diameter??

Thank you so much for any help!

**Comment by terjeio** [comment]
> Should the steps/ degree be the same for any diameter workpiece? 

Yes, if you command a 360 degree move then that is one turn of the rotary. The surface distance will of course depend on the diameter. You have to translate mm moves on the surface to degrees - your CAM program can do that for you? IIRC there are also specialized programs available that can "wrap" a linear axis to a rotary.
ioSender can also do so, but this is untested functionality - see the _File > Transform_ submenu. Note that the wrap menuitem is only available if more than 3 axes are configured.

**Reply by Gabjohn1** [reply]
Ok I use Vetric Aspire.  It has a wrapping feature.  I will run a couple files and test it.
Thank You!!

**Comment by Toutoune86** [comment]
>  ioSender can also do so, but this is untested functionality - see the _File > Transform_ submenu. Note that the wrap menuitem is only available if more than 3 axes are configured.


Hello !
I am just new into IoSender. Thank you very much Terje for your awesome work ! I am planning to make a rotary plasma cutter for tubes
I configured my teensy firmware to have four axis, but still can not access to the Transform submenu, in File menu ; is there another step I missed ?

Another more fundamental question which may be naïve as it looks like both of you implicitly replyed "no" : Is there no way to define  the rotary axis unit in degrees instead of length ? No alternative to redefining for each different diameter the steps/mm  (ie $103) ?

**Reply by terjeio** [reply]
> I configured my teensy firmware to have four axis, but still can not access to the Transform submenu, in File menu ; is there another step I missed ?

You have to load gcode before the menu is enabled.

> Is there no way to define the rotary axis unit in degrees instead of length ?

Set $376 (in the `Stepper` category) and restart the sender.

**Reply by Toutoune86** [reply]
Thank you !

**Reply by Toutoune86** [reply]
There is no 3d rendering of a wrapped axis I guess. At least I see nothing, everything beeing compressed along the X axis. But seing both the A and X DRO changing during this simulation is awesome ! it looks like actual tube notching !
It is a  very little step for humanity, but a big one for me, thank you !
